src/common/mfu_flist_walk.c

```diff
@@ -656,9 +669,6 @@ static void walk_stat_process(CIRCLE_handle* handle)
 
     /* TODO: filter items by stat info */
 
-    /* record info for item in list */
-    mfu_flist_insert_stat(CURRENT_LIST, path, st.st_mode, &st);
-
```

Comment:
Can't think of anything off the top of my head, but I think it'd be good to keep the ordering here where we insert the directory into the list before we recurse into it.